runtime/bindings/python/src/pyopenvino/core/async_infer_queue.cpp

```diff
@@ -40,7 +40,8 @@ class AsyncInferQueue {
         _cv.wait(lock, [this] {
             return !(_idle_handles.empty());
         });
-
+        if (_errors.size() > 0)
+            throw _errors.front();
```

Comment:
Is it safety to throw python exception without GIL?

Reply:
Actually it is a good question. I found only one example where exceptions are thrown in more _sophisticated_ way from different threads and the GIL was released before each throw. However we can put it under more investigation later on. I could not think about any reproducer for this case...